staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -384,8 +385,15 @@ func NewCacherFromConfig(config Config) (*Cacher, error) {
 		<-cacher.timer.C
 	}
 
+	sizeFunc := func(obj runtime.Object) (int, error) {
+		w := &bytes.Buffer{}
+		err := config.Codec.Encode(obj, w)
```

Comment:
You pointed me to another attempt:
https://github.com/kubernetes/kubernetes/compare/master...mborsz:kubernetes:storage_bytes_bad?expand=1#diff-6c00e4dbc36a123cb4ad0828c8aa43d026159563a7fbd1887a3445d6c341dd47R463

Can we maybe do some hybrid of that, so that:
- for watch events we take the data from etcd
- for lists (where we don't know how to easily pass that) - we compute it?
Re-listing should be rare, so it sounds more acceptable to me...